### PR TITLE
Split "setting up development" into a separate guide

### PR DESCRIPTION
**Pull request by MatMoore**
Move this from `CONTRIBUTING.md` to `docs/setting-up-development.md`, so that it's 
easier to understand and link to.

I've also reordered some of the sections in it to make it easier to get to the point where you have the app running.

See issue https://github.com/exercism/exercism.io/issues/3251

### CODE REVIEW

File: docs/setting-up-local-development.md [hunk 1]
```diff
@@ -1 +1,286 @@
...
+Backend:
+
+- Ruby
+- PostgreSQL
+
+Frontend:
+
+- Node.js
+
+To install Ruby, check out [RVM](https://rvm.io), [rbenv](https://github.com/sstephenson/rbenv) or [ruby-install](https://github.com/postmodern/ruby-install).
```
Comment (ErikSchierboom) [inline docs/setting-up-local-development.md]: Perhaps you could link directly to the installation documentation pages? For example, for RVM you could refer to https://rvm.io/rvm/install.

File: docs/setting-up-local-development.md [hunk 2]
```diff
@@ -1 +1,286 @@
...
+
+PostgreSQL can be installed with [Homebrew](http://brew.sh) on Mac OS X: `brew install postgresql`
```
Comment (ErikSchierboom) [inline docs/setting-up-local-development.md]: Two small things:

1. Perhaps you could make it a bulleted list, with an entry for each platform. 
2. I'd like to see a link to a Windows installation too.

File: docs/setting-up-local-development.md [hunk 3]
```diff
@@ -1 +1,286 @@
...
+Note that you may have trouble running commands like `rake` or `pry` if your system has a different version of those gems than the Gemfile. In that case, you will need to prepend those commands with `bundle exec` so that bundler will execute that command in the context of the project's gems rather than your system's gems. If you would like to learn more about this topic and how to alias `bundle exec` (if you don't like typing it out each time), see [this article by Thoughtbot](https://robots.thoughtbot.com/but-i-dont-want-to-bundle-exec).
+
+To debug the database setup, do it step-by-step:
+
+* Create the PostgreSQL database: `rake db:setup`
+* Run the database migrations: `rake db:migrate`
+* Fetch the seed data: `rake db:seeds:fetch`
+* Seed the database: `rake db:seed`
+
+If you are having 'PG::Connection ...' or 'Peer authentication failed for user ...' issues you can follow these steps (this is assuming that you modified the database.yml file and your changes are not being accessed properly. Anywhere that it is stated "PGUSER=postgres" replace "postgres" with your specified username that you use for your postgresql databases):
```
Comment (ErikSchierboom) [inline docs/setting-up-local-development.md]: I would put the error messages in backticks, to make them standout.

**Comment by kytrinyx** [comment]
I'm stoked to see this change—the Contributing Guide has been an overwhelming wall of text and this definitely helps.

**Comment by MatMoore** [comment]
Thanks for the review @ErikSchierboom. I've added a few more commits implementing your suggestions.

I've linked to [https://www.postgresql.org/download/windows/](https://www.postgresql.org/download/windows/) which has installers for windows (I haven't tried these myself but it seems like the obvious way to do it). For everything else we're already linking out to instructions that cover multiple platforms so I don't think I can improve on that.

I also split up the troubleshooting section a bit more. I think it's still a bit confusing which `db:` tasks do what and why, but hopefully that helps.

**Comment by ErikSchierboom** [comment]
Great, LGTM!

**Comment by kytrinyx** [comment]
Thank you, @MatMoore!

**Pull request merged**